File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/user/dto/UserNewPasswordRequest.java
package com.portfolio.auctionmarket.domain.user.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class UserNewPasswordRequest {

    @NotBlank(message = "현재 비밀번호를 입력해 주세요")
    @Size(min = 8, message = "비밀번호는 최소 8자 이상이어야 합니다")
    private String currentPassword;

    @NotBlank(message = "변경할 비밀번호를 입력해 주세요")
    @Size(min = 8, message = "비밀번호는 최소 8자 이상이어야 합니다")
    private String newPassword;

    @NotBlank(message = "변경할 비밀번호를 한 번 더 입력해 주세요")
    @Size(min = 8, message = "비밀번호는 최소 8자 이상이어야 합니다")
    private String confirmPassword;


}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/user/dto/UserNewPasswordRequest.kt
package org.example.auctioncommon.domain.user.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class UserNewPasswordRequest(
    @field:NotBlank(message = "현재 비밀번호를 입력해 주세요")
    @field:Size(min = 8, message = "비밀번호는 최소 8자 이상이어야 합니다")
    val currentPassword: String,

    @field:NotBlank(message = "변경할 비밀번호를 입력해 주세요")
    @field:Size(min = 8, message = "비밀번호는 최소 8자 이상이어야 합니다")
    val newPassword:  String,

    @field:NotBlank(message = "변경할 비밀번호를 한 번 더 입력해 주세요")
    @field:Size(min = 8, message = "비밀번호는 최소 8자 이상이어야 합니다")
    val confirmPassword: String
) {

}
